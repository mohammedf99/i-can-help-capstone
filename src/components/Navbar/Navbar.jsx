import React from 'react'

export const Navbar = () => {
  return (
    <div>
      This is an example of a component
    </div>
  )
}
