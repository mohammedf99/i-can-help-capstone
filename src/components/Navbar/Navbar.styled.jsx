/* eslint-disable import/prefer-default-export */
import styled from "styled-components";

export const Navbar = styled.div`
  .Navbar {
    font-family: Roboto;
    color: #1c1259;
    min-height: 10vh;
    background: FFFFFF;
    box-shadow: 0px 1px 10px #aaa;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
  }
  .site-logo {
    margin-left: 35px;
    width: 20%;
  }
  .site-logo h1 {
    color: #1c1259;
    font-style: normal;
    font-size: 36px;
    line-height: 20px;

    &:hover {
      cursor: pointer;
    }
  }

  .Searchbtn {
    border: 1px solid #1c1259;
    box-sizing: border-box;
    border-radius: 40px;

    width: 263px;
  }
  .ant-input-lg {
    color: #1c1259;
  }
  .nav {
    display: flex;
    justify-content: space-between;
    padding: 0 4rem 0 0;
    margin-top: 30px;
    width: 55%;
    color: #1c1259;
  }
  .btn {
    width: 30%;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
  }
  .btn1 {
    margin-right: 5px;
    background: #ffffff;
    border: 1px solid #1c1259;
    transition: all 0.4s;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: row-reverse;
  }
  .btn1:hover {
    background: #1c1259;
    box-shadow: 0px 10px 20px rgba(0, 0, 0, 0.2);
    border-radius: 40px;
    width: 30%;
  }
  .btn1:hover::after {
    content: "Login";
    color: white;
    border: 1px solid black;
  }
  .btn2 {
    margin-right: 5px;
    background: #ffffff;
    border: 1px solid #1c1259;
    transition: all 0.4s;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: row-reverse;
  }

  .btn2:hover {
    background: #1c1259;
    box-shadow: 0px 10px 20px rgba(0, 0, 0, 0.2);
    border-radius: 40px;
    width: 30%;
  }

  .btn2:hover::after {
    content: "Sign Up";
    color: white;
    border: 1px solid black;
  }

  .arrow {
    transition: all 0.2s;
  }

  .arrow:hover {
    transform: rotate(-90deg);
    color: white;
    display: flex;
    flex-direction: row-reverse;
  }

  .navcontent {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
  }

  .navconntent:first-child,
  .navcontent:last-child {
    margin: 0;
  }
  .navcontent a {
    margin: 0 1rem;
  }

  .navcontent a {
    display: flex;
    color: #1c1259;
    justify-content: space-evenly;
    font-family: Roboto;
    font-style: normal;
    font-weight: 500;
    font-size: 18px;
    line-height: 20px;
  }
  .anticon-search {
    margin-top: 5px;
  }
  .avatar:hover {
    cursor: pointer;
  }
`;
