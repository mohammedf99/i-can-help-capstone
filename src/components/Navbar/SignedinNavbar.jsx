import React, { useContext } from "react";

import { Avatar } from "antd";

import { createFromIconfontCN } from "@ant-design/icons";
import { signout } from "../../Utilities/FirebaseUtilities";
import { AuthContext } from "../../Utilities/auth/Auth";


const IconFont = createFromIconfontCN({
  scriptUrl: "//at.alicdn.com/t/font_8d5l8fzk5b87iudi.js",
});

const SignedinNavbar = () => {
  const { userData } = useContext(AuthContext);
  return (
    <div className="btn">
      <Avatar className="avatar" src={userData?.picture} size={35} />

      <Avatar
        className="avatar"
        size={35}
        icon={<IconFont type="icon-tuichu" size="21.25px" />}
        style={{ backgroundColor: "#EE4266", marginLeft: "15px" }}
        onClick={() => signout()}
      />
    </div>
  );
};


export default SignedinNavbar;
