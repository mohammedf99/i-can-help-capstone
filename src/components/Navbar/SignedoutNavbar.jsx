import React from "react";

import { Button } from "antd";

import { DownOutlined, UserOutlined } from "@ant-design/icons";

const SignedoutNavbar = ({ auth }) => (
  <div className="btn">
    <Button
      className="btn1"
      type="button"
      shape="circle"
      icon={<UserOutlined />}
      onClick={() => auth.changeSigninVisible()}
    />

    <Button
      className="btn2"
      type="button"
      shape="circle"
      icon={<DownOutlined className="arrow" />}
      onClick={() => auth.changeSignupVisible()}
    />
  </div>
);

export default SignedoutNavbar;
