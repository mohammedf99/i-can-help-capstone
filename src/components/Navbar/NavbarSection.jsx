import React, { useContext } from "react";
import { useTranslation } from "next-i18next";
import { Input } from "antd";
import Link from "next/link";
import { SearchOutlined } from "@ant-design/icons";
import { useRouter } from "next/router";
import { Navbar } from "./Navbar.styled";
import SignedinNavbar from "./SignedinNavbar";
import SignedoutNavbar from "./SignedoutNavbar";
import AuthContext from "../../Utilities/Contexts/AuthContext";

const NavbarSection = ({ transparent, auth }) => {
  const user = useContext(AuthContext).currentUser;
  
  const { t } = useTranslation("navbar");
  
  const router = useRouter();
  
  return (
    <Navbar>
      <div
        className="Navbar"
        style={
          transparent && { backgroundColor: "transparent", boxShadow: "none" }
        }
      >
        <div className="site-logo">
          <Link href="/">
            <h1>I Can Help</h1>
          </Link>
        </div>

        <div className="nav" style={{ margin: 0 }}>
          {user && (
            <div className="navcontent">
              <Link href="/home">{t("post")}</Link>
              <Link href="/myPosts">{t("myPosts")}</Link>
              <Link href="/pinnedPosts">{t("pinnedPost")}</Link>
            </div>
          )}

          {!user && (
            <div className="navcontent">
              <Link href="/about">{t("about")}</Link>
              <a
                onClick={() =>
                  router.push({ pathname: "/search", query: { type: 1 } })
                }
              >
                {t("findWork")}
              </a>
              <a
                onClick={() =>
                  router.push({ pathname: "/search", query: { type: 2 } })
                }
              >
                {t("findTalent")}
              </a>
            </div>
          )}

          <Input
            className="Searchbtn"
            size="large"
            placeholder={t("searchHere")}
            suffix={<SearchOutlined />}
            onPressEnter={(e) =>
              router.push({ pathname: "/search", query: { q: e.target.value } })
            }
          />
        </div>

        {!user ? <SignedoutNavbar auth={auth} /> : <SignedinNavbar />}
      </div>
    </Navbar>
  );
};

export default NavbarSection;
