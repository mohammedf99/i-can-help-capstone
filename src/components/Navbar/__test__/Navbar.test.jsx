import * as React from "react";
import * as renderer from "react-test-renderer";
import Navbar from "../NavbarSection";

it("renders Navbar correctly", () => {
  const tree = renderer.create(<Navbar />).toJSON();
  expect(tree).toMatchSnapshot();
});
