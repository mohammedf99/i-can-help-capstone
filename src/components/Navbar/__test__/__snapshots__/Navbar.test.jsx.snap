// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar correctly 1`] = `
<div>
  This is an example of a component
</div>
`;
